profiles/common/modules/features/cce_basic_config/cce_basic_config.module

```diff
@@ -341,3 +360,54 @@ function cce_basic_config_modules_enabled($modules) {
     multisite_config_service('user')->grantPermission('editor', array('export nodes'));
   }
 }
+
+/**
+ * Helper function to validate a file's mime type.
+ */
+function _cce_basic_config_validate_real_mimetype(&$errors, $file) {
+  if (variable_get("ec_enable_real_mimetype_validation", TRUE) && !_cce_basic_config_is_cron_run()) {
+    // Get mime type from filename.
+    $mime_by_filename = file_get_mimetype($file->filename);
+    // Get mime type from file contents.
+    $finfo = finfo_open(FILEINFO_MIME_TYPE);
+    // Drupal will in some specific cases append a .txt extension to a file
+    // which will mess with the temporary PHP uri.
+    $temporary_uri = explode(".", $file->uri);
+    $real_mime = finfo_file($finfo, $temporary_uri[0]);
+    if ($mime_by_filename !== $real_mime) {
+      $errors[] = "The file's content type doesn't match the extension.";
+    }
+  }
+}
+
+/**
+ * Helper function to check a mimetype against a whitelist.
+ */
+function _cce_basic_config_validate_allowed_mimetype(&$errors, $file) {
+  $allowed_mimetypes = variable_get("ec_mimetype_whitelist", array());
+  if (!empty($allowed_mimetypes)) {
+    $mimetype = file_get_mimetype($file->filename);
+    if (!in_array($mimetype, $allowed_mimetypes)) {
+      $errors[] = "The file type is not allowed.";
+    }
+  }
+}
+
+/**
+ * Helper function to check if a user has access to the destination file system.
+ */
+function _cce_basic_config_validate_file_destination(&$errors, $file) {
+  if (variable_get("ec_enable_file_destination_validation", TRUE)) {
+    $destination_file_scheme = file_uri_scheme($file->destination);
+    if ($destination_file_scheme !== "private" && user_is_anonymous()) {
+      $errors[] = "You don't have permissions to upload files to this folder.";
```

Comment:
I suggest to mention that if the problem persists, the user must contact the administrator.

I think also that we should log the error (watchdog).